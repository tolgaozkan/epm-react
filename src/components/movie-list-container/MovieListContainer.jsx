import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import MovieList from '../movie-list/MovieList';

import styles from './style.css';

export class MovieListContainer extends React.PureComponent {
  render() {
    const { movies } = this.props;
    return (
      <div className={styles['movie-list-container']}>
        <MovieList movies={movies} />
      </div>
    );
  }
}

MovieListContainer.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

function mapStateToProps(state) {
  return {
    movies: state.movies,
  };
}

export default connect(mapStateToProps)(MovieListContainer);
