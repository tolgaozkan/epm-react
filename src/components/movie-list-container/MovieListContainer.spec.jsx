import * as React from 'react';
import { create as render } from 'react-test-renderer';

import MovieListContainer from './MovieListContainer';

class MockResponse {
  json() {
    return {
      data: [],
    };
  }
}

beforeEach(() => {
  global.fetch = jest.fn().mockImplementation(() => {
    const p = new Promise((resolve) => {
      resolve(new MockResponse());
    });

    return p;
  });
});

it('renders movie list container', () => {
  const result = render(<MovieListContainer />);
  const snap = result.toJSON();

  expect(snap).toMatchSnapshot();
});
