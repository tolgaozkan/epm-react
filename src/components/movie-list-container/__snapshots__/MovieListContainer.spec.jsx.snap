// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders movie list container 1`] = `
<div
  className="movie-list-container"
/>
`;
